qrcodereaderview/src/main/java/com/dlazaro66/qrcodereaderview/QRCodeReaderView.java

```diff
@@ -132,34 +139,37 @@ public CameraManager getCameraManager() {
   // Called when camera take a frame
   @Override public void onPreviewFrame(byte[] data, Camera camera) {
 
-    PlanarYUVLuminanceSource source =
-        mCameraManager.buildLuminanceSource(data, mPreviewWidth, mPreviewHeight);
-
-    HybridBinarizer hybBin = new HybridBinarizer(source);
-    BinaryBitmap bitmap = new BinaryBitmap(hybBin);
-
-    try {
-      Result result = mQRCodeReader.decode(bitmap);
-
-      // Notify we found a QRCode
-      if (mOnQRCodeReadListener != null) {
-        // Transform resultPoints to View coordinates
-        PointF[] transformedPoints = transformToViewCoordinates(result.getResultPoints());
-        mOnQRCodeReadListener.onQRCodeRead(result.getText(), transformedPoints);
-      }
-    } catch (ChecksumException e) {
-      Log.d(TAG, "ChecksumException");
-      e.printStackTrace();
-    } catch (NotFoundException e) {
-      // Notify QR not found
-      if (mOnQRCodeReadListener != null) {
-        mOnQRCodeReadListener.QRCodeNotFoundOnCamImage();
+    if (qrDecodingEnabled.get()) {
```

Comment:
1. This method gets executed in Main Thread so you don't need it to be AtomicBoolean
2. It's more readable if you do:

```java
@Override public void onPreviewFrame(byte[] data, Camera camera) {
   if (!qrDecodingEnabled) {
      return;
   }

   // do something
}
```